keanu-python/tests/test_sampling.py

```diff
@@ -53,3 +56,45 @@ def test_down_sample_interval(net):
 
     expected_num_samples = draws / down_sample_interval
     assert all(len(vertex_samples) == expected_num_samples for vertex_id, vertex_samples in samples.items())
+
+
+@pytest.mark.parametrize("algo", [
+    ("metropolis"),
+    ("hamiltonian")
+])
+def test_iter_is_correct_size(algo, net):
+    draws = 10
+    samples = iterate_samples(net=net, sample_from=net.get_latent_vertices(), algo=algo, draws=draws, down_sample_interval=1)
+    assert sum(1 for i in samples) == 10
+
+
+@pytest.mark.parametrize("algo", [
+    ("metropolis"),
+    ("hamiltonian")
+])
+def test_iter_returns_same_result_as_sample(algo):
+    draws = 100
+    model = Thermometer.model()
+    net = BayesNet(model.temperature.get_connected_graph())
+    set_starting_state(model)
+    samples = sample(net=net, sample_from=net.get_latent_vertices(), algo=algo, draws=draws)
+    set_starting_state(model)
+    iter_samples = iterate_samples(net=net, sample_from=net.get_latent_vertices(), algo=algo, draws=draws)
+    iter_samples_dict = {}
+
+    for next_sample in iter_samples:
+        for vertex_id, vertex_sample in next_sample.items():
+            if vertex_id in iter_samples_dict:    
+                iter_samples_dict[vertex_id] += vertex_sample
+            else:
+                iter_samples_dict[vertex_id] = vertex_sample
+
+    for vertex_id, summed_vertex_sample in iter_samples_dict.items():
+        average = summed_vertex_sample / draws
+        np.testing.assert_almost_equal(average, np.average(samples[vertex_id]))
+
+def set_starting_state(model):
+    KeanuRandom().set_default_random_seed(1)
```

Comment:
`KeanuRandom.set_default_random_seed(1)`. Also make sure to make that method static.